### ISSUE TITLE
Multiple scenarios (define scenarios as array)

### ISSUE THREAD

**Issue by khvalov** [yiisoft/yii2#16731]
Is any plans to get implemented defining multiple validation scenarios like: 
 $model->scenario = [Shippings::SCENARIO_ONE,Shippings::SCENARIO_TWO]; 

Reason: 
I have a complex matrix when some **group of fields**  needs to be validated and when it's not. The most convenient way to get it implemented - define scenarios and add it as array based on designed scenario

**Comment by rob006** [comment]
> I have a complex matrix when some **group of fields** needs to be validated and when it's not. The most convenient way to get it implemented - define scenarios and add it as array based on designed scenario

Sounds like you're overusing scenarios for cases where you should have separated form model. Or some "rules builder" at least.

**Comment by khvalov** [comment]
Any ideas what is better approach to implement next business logic: 
- I have shipping with sender/receiver addresses (country/city/address)
- I have workflow in place when some 3rd party may pre-populate some fields like receiver address (in different Model) 
Workflow is like: 
- If we have prepopulated addresses - I have to ask user choose from prefilled 
- If we do not - I have to ask full address. 

Imho the best way to use scenarios and push it to array. Like: 
- If we have prepopulated address - > push "prepoulated" scenario. If we are not - push "full address" validation scenario. 

Any thoughts ?
